### PR TITLE
fix handling of flags 1-3 in coons shading

### LINKED ISSUE #4227: Coons patch mesh gradient displays white areas
http://www.tcpdf.org/examples/example_030.pdf

#4192 mostly fixed this, but the second page still has some problems (white areas).

Another test file:

http://tavmjong.free.fr/SVG/MESH/lamp_cairo.pdf

/cc @yurydelendik

### PR DESCRIPTION
**Pull request by CodingFabian**
Short story: somebody got lost in two different indices. pi is an index in the stream and is explained on page 198 of the 32000-spec (however 1-based there), and ps is an index to something in PDF.js. I used the code from flag 0 (which works) to understand which is which. It is also important to understand that for flags 1,2 and 3, the stream is always assigned to the same coordinates and colors. What changes is which "old" coordinates and colors are assigned to what is "missing" in the stream. This is why for these flags, the code is identical except for the assignments in the first "row".

Largely fixes #4227.

### CODE REVIEW

File: test/test_manifest.json [hunk 1]
```diff
@@ -2247,6 +2247,12 @@
        "lastPage": 4,
        "type": "eq"
     },
+    {  "id": "issue4227",
+      "file": "pdfs/coons-allflags-withfunction.pdf",
+      "md5": "c5f79c24bf9eb66698be0e4ecaa1bdf8",
+      "rounds": 1,
+      "type": "eq",
```
Comment (yurydelendik) [inline test/test_manifest.json]: nit: extra comma (bad in JSON)

Comment (timvandermeij) [inline test/test_manifest.json]: Also the indenting is a bit off. Compare with the ones below.

Comment (CodingFabian) [inline test/test_manifest.json]: huh? shows identical in my editor

Comment (CodingFabian) [inline test/test_manifest.json]: ah. lol. perhaps i copied the only other bad one in this file :)

Comment (CodingFabian) [inline test/test_manifest.json]: actually the whole file is an indentation mess

Comment (timvandermeij) [inline test/test_manifest.json]: You're right. Keep it like https://github.com/CodingFabian/pdf.js/blob/fixup-6303/test/test_manifest.json#L2331-L2336 for this PR as that is what we always use.

Comment (CodingFabian) [inline test/test_manifest.json]: i can fixup the indention in this file, but which one is the wanted one. give me a reference test case and ill do it now

Comment (timvandermeij) [inline test/test_manifest.json]: We should do that in a different PR. I actually do not like them all myself, and I would prefer:

<pre>
[
  {
    "id": "foo",
    "name": "bar"
  }
]
</pre>

which is also how JSONLint formats it (and a lot of other projects).

**Comment by THausherr** [comment]
> looks much better now (but there is still work to do to make your sample render as it should :-))
Now I see what you mean (I had tested my patch on the CIB file, not on the file I created today). This looks like the effect that happens also with the lamp_cairo.pdf file. (I have a patch for type 7, and it works fine for another test file, but not for lamp_cairo.pdf). I have the suspicion that this is related to the effect of "extra stuff" that has been seen with types 4 and 5.

**Comment by yurydelendik** [comment]
Thank you for the patch.

/botio makeref
